src/main/thrift/AromaService.thrift

```diff
@@ -614,85 +547,230 @@ struct GetReactionsResponse
+
+struct GetDashboardRequest
 {
     1: UserToken token;
-    /** Performs a search based on the Application name. */
-    2: string applicationName;
-    3: optional uuid organizationId;
 }
 
-struct SearchForApplicationsResponse
+struct GetDashboardResponse
 {
-    1: list<Application> applications = []
+    1: int unreadMessageCount = 0;
+    2: int totalMessagesLastHour = 0;
+    3: int totalMessagesLast24hrs = 0;
+    4: list<Aroma.Message> recentMessages = [];
+    5: int numberOfLowUrgencyMessages = 0;
+    6: int numberOfMediumUrgencyMessages = 0;
+    7: int numberOfHighUrgencyMessages = 0;
 }
 
+//==========================================================
+// DEVICE REGISTRATION OPERATIONS
+//==========================================================
 
-/**
- * The Aroma Service is designed to be used by People, and not Applications.
- * They provide query interfaces and Authentication/Authorization over Data.
- */
-service AromaService
+struct CheckIfDeviceIsRegisteredRequest
 {
+    1: UserToken token;
+    2: MobileDevice device;
+}
 
-    double getApiVersion()
+struct CheckIfDeviceIsRegisteredResponse
+{
+    1: required bool isRegistered;
+}
+
+/**
+ * Get a list of devices registered to the calling user.
+ */
+struct GetRegisteredDevicesRequest
+{
+    1: UserToken token;
+}
+
+struct GetRegisteredDevicesResponse
+{
+    1: list<MobileDevice> devices = [];
+}
+
+struct RegisterDeviceRequest
+{
+    1: UserToken token;
+    2: MobileDevice device;
+}
+
+struct RegisterDeviceResponse
+{
+    1: optional string message = "Successfully Registered";
+}
+
+struct UnregisterDeviceRequest
+{
+    1: UserToken token;
+    2: MobileDevice device;
+}
+
+struct UnregisterDeviceResponse
+{
+    1: MobileDevice removedDevice;
+}
+
+
+//==========================================================
+// OTHER OPERATIONS
+//==========================================================
+
+/**
+ * Registers an Application Endpoint to use
+ * for Health Checks.
+ */
+struct RegisterHealthCheckRequest
+{
+    1: UserToken token;
+    2: Endpoint endpoint;
+}
+
+struct RegisterHealthCheckResponse
+{
+    1: optional string message;
+    2: optional string healthCheckToken;
+}
+
+
+/**
+ * Request to get any upcoming Service Announcements
+ * from the Team @ Aroma Tech.
+ */
+struct GetServiceAnnouncementsRequest
+{
+    1: UserToken token;
+}
+
+struct GetServiceAnnouncementsResponse
+{
+    1: optional list<Aroma.ServiceAnnouncement> serviceAnnouncements = []
+}
+
+
+//==========================================================
+// AROMA SERVICE DEFINITION
+//==========================================================
+
+/**
+ * The Aroma Service is designed to be used by People, and not Applications.
+ * They provide query interfaces and Authentication/Authorization over Data.
+ */
+service AromaService
+{
 
     //==========================================================
-    // Action Operations
+    // AUTHENTICATION OPERATIONS
```

Comment:
It's better to do this kind of refactoring in another branch, but I'll let it slide for now 😎